src/app/problems/problem-create/mcq-create-snippet/mcq-create-snippet.component.ts

```diff
@@ -74,34 +83,52 @@ export class McqCreateSnippetComponent implements OnInit {
         }
     }
 
-    onSubmit(formData: FormGroup): void {
+    /**
+     * Form submission.
+     */
+    onSubmit(): void {
+        const data = McqForm.extractData(this.formGroup);
         let submissionRequest;
         if (!this.checkBox) {
-            submissionRequest = this.problemHelpersService.createMCQSubmissionRequest(formData.value, this.distractors.map(x => x.text), this.variables, this.questionText, this.answerText);
+            submissionRequest = this.problemHelpersService.createMCQSubmissionRequest(data, this.distractors.map(x => x.text), this.variables, this.questionText, this.answerText);
         } else if (this.checkBox) {
-            submissionRequest = this.problemHelpersService.createCheckboxSubmissionRequest(formData.value, this.distractors.map(x => x.text), this.variables, this.questionText, this.correctAnswers.map(x => x.text));
+            submissionRequest = this.problemHelpersService.createCheckboxSubmissionRequest(data, this.distractors.map(x => x.text), this.variables, this.questionText, this.correctAnswers.map(x => x.text));
```

Comment:
These functions should go in the forms now. When you extract data it should give an object that can be sent in the API.